frontend/test/metabase-lib/lib/queries/StructuredQuery.unit.spec.js

```diff
@@ -338,6 +338,17 @@ describe("StructuredQuery", () => {
         expect(query.breakout(ORDERS.PRODUCT_ID).hasValidBreakout()).toBe(true);
       });
     });
+
+    describe("excludes breakout that has the same base dimension as what is already used", () => {
+      const breakout = [
+        "field",
+        ORDERS.CREATED_AT.id,
+        { "temporal-unit": "month" },
+      ];
+      const queryWithBreakout = query.breakout(breakout);
+      expect(queryWithBreakout.breakoutOptions().all().length).toBe(27);
+      expect(queryWithBreakout.breakoutOptions(breakout).all().length).toBe(28);
```

Comment:
I'm worried the `length` check might not be sufficient enough (what if we miss another dimension we actually expect instead of `CREATED_AT` and the length is the same? I'd look into a way to assert `CREATED_AT` is not included

Reply:
Good call! I'll find a way to look through the dimensions